### PR TITLE
[cmake] update tests definitions to match those run by CI

### PR DESCRIPTION
**Pull request by Fat-Zer**
Update cmake's tests definitions to match those currently in CI system, as mentioned in #217.

PS: I've already did this pretty much right after my previous message, and attaching it so it won't get lost accidentally. Feel free to reject it if it's not appropriate.

### CODE REVIEW

**Comment by K1ngfish3r** [comment]
for the following tests, max-diff value is different
```
"--style=gradient-radial"        , "--max-diff=2",
"--style=gradient-radial-dither" , "--max-diff=2",
"--style=gradient-conic"         , "--max-diff=2",
"--style=gradient-conic-dither"  , "--max-diff=2",
```
all the other tests build fine using this patch

**Comment by kobalicek** [comment]
I'm not sure I want to replicate the CI logic in cmake, which is already rather long and adding more to it is not my preference.

I think maybe changing the context testers in a way to do most stuff when executed without arguments (or adjusting defaults) would be better. The reason CI invokes some files multiple times is to get exactly which one failed without having to deep dive into logs, etc... In addition, since CI runs many configurations including sanitizers context testers use parameters to lower the number of iterations of each test to a reasonable value that works with sanitizers and emulation, however, if you do local testing you probably want these values increased, etc...

File: CMakeLists.txt [hunk 1]
```diff
@@ -1124,7 +1124,26 @@ if (NOT BLEND2D_EMBED)
...
+    add_test(NAME bl_test_context_simple
+             COMMAND bl_test_context_simple "--count=100" "--comp-op=all" "--style=random")
+
+    foreach(style solid gradient-linear gradient-linear-dither gradient-radial
+                  gradient-radial-dither gradient-conic gradient-conic-dither
+                  pattern-aligned pattern-fx pattern-fy pattern-fx-fy
+                  pattern-affine-nearest pattern-affine-bilinear)
+      add_test(NAME "bl_test_context_jit-${style}"
+               COMMAND bl_test_context_jit "--count=200" "--simd-level=all" "--comp-op=all"
+                       "--opacity-op=all" "--style=${style}" "--max-diff=0" "--quiet")
```
[the remark this answers is not in the document]
Comment (Fat-Zer) [reply CMakeLists.txt]: good catch, I've modified it a bit so it won't look so bulky.

**Comment by Fat-Zer** [comment]
> I'm not sure I want to replicate the CI logic in cmake, which is already rather long and adding more to it is not my preference.

You can neatly move all the testing stuff into a separate file. Also you may do the same with other stuff like utility functions and some tests.

Besides that take into account, that having tests bundled with source code is very useful for distribution packagers as it will help catch bugs caused by unusual configurations. The library doesn't have a lot of dependencies, but there are always different compilers and aggressive optimization flags.


PS: if you decide not to go ahead with this PR, could you please at least remove  #217

**Comment by Fat-Zer** [comment]
Closing in favour of solution introduced in d2e936a0fc2be39363b2922aa50a90378190dea1

**Pull request closed without merging**